src/workflow_provider/default_vm_workflow_provider.cpp

```diff
@@ -0,0 +1,376 @@
+                {
+                    mpl::log(
+                        mpl::Level::error, category,
+                        fmt::format("Invalid workflow name \'{}\': must be a valid host name", file_info.baseName()));
+                    needs_update = true;
+
+                    continue;
+                }
+
+                Poco::Zip::ZipInputStream zip_input_stream{zip_stream, it->second};
+                std::ostringstream out(std::ios::binary);
+                Poco::StreamCopier::copyStream(zip_input_stream, out);
+                workflows_map[file_info.baseName().toStdString()] = YAML::Load(out.str());
+            }
+        }
+    }
+
+    return workflows_map;
+}
+} // namespace
+
+mp::DefaultVMWorkflowProvider::DefaultVMWorkflowProvider(const QUrl& workflows_url, URLDownloader* downloader,
+                                                         const QDir& archive_dir,
+                                                         const std::chrono::milliseconds& workflows_ttl)
+    : workflows_url{workflows_url},
+      url_downloader{downloader},
+      archive_file_path{archive_dir.filePath(github_workflows_archive_name)},
+      workflows_ttl{workflows_ttl}
+{
+    try
+    {
+        update_workflows();
+    }
+    catch (const std::exception& e)
+    {
+        mpl::log(mpl::Level::error, category, fmt::format("Error on workflows start up: {}", e.what()));
+    }
+}
+
+mp::DefaultVMWorkflowProvider::DefaultVMWorkflowProvider(URLDownloader* downloader, const QDir& archive_dir,
+                                                         const std::chrono::milliseconds& workflows_ttl)
+    : DefaultVMWorkflowProvider(default_workflow_url, downloader, archive_dir, workflows_ttl)
+{
+}
+
+mp::Query mp::DefaultVMWorkflowProvider::fetch_workflow_for(const std::string& workflow_name,
+                                                            VirtualMachineDescription& vm_desc)
+{
+    update_workflows();
+
+    Query query{"", "", false, "", Query::Type::Alias};
+    auto& workflow_config = workflow_map.at(workflow_name);
+
+    auto workflow_instance = workflow_config["instances"][workflow_name];
+
+    if (workflow_instance["image"])
+    {
+        // TODO: Support http later.
+        // This only supports the "alias" and "remote:alias" scheme at this time
+        auto image_str{workflow_config["instances"][workflow_name]["image"].as<std::string>()};
+        auto tokens = mp::utils::split(image_str, ":");
+
+        if (tokens.size() == 2)
+        {
+            query.remote_name = tokens[0];
+            query.release = tokens[1];
+        }
+        else if (tokens.size() == 1)
+        {
+            query.release = tokens[0];
+        }
+        else
+        {
+            needs_update = true;
+            throw InvalidWorkflowException("Unsupported image scheme in Workflow");
+        }
+    }
+
+    if (workflow_instance["limits"]["min-cpu"])
```

Comment:
What if `["limits"]` isn't there?

Reply:
I'm pretty sure it behaves exactly how it does now, but I'll make sure and add a test workflow that does not define a `limits` section.

Reply:
Yeah, if `limits` is not defined, this still picks that up.  I added a test workflow to cover this particular case.